app/src/main/java/com/yalta/viewmodel/MapViewModel.kt

```diff
@@ -58,4 +72,13 @@ class MapViewModel(application: Application) : AndroidViewModel(application) {
     private fun unsubscribe() {
         fusedLocationProviderClient.removeLocationUpdates(locationCallback)
     }
-}
\ No newline at end of file
+}
+
+class MapViewModelFactory(
+    private val application: Application
+) :
+    ViewModelProvider.Factory {
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        return MapViewModel(application) as T
```

Comment:
Ok, what do we have here... java heritage? 
1) Why `create` is parametrized if we basically only work with `MapViewModel`?
2) I would guess that this should be more generic, right? I mean we do not want to write same factory for each Viewmodel, right? It should be `YaltaViewMovelFactory` or something like that, right? But we can't call constructor of `T` cause erasure? Or we can since we have class param? Or we can do that with `reified`? or we can't cause we overload some java code? 
So many questions... 
But main idea is that this should be generalized: 
```
class YaltaViewModelFactory(
    private val application: Application
) :
    ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return modelClass.newInstance(application)
    }
}
```
I would guess it would look something like this and work for every type of VM that has application constructor param